gnd/src/main/java/com/google/android/gnd/persistence/local/room/RoomLocalDataStore.java

```diff
@@ -340,24 +341,26 @@ public Completable mergeFeature(Feature feature) {
         .subscribeOn(schedulers.io());
   }
 
+  @SuppressLint("TimberArgCount")
   @Transaction
   @Override
   public Completable mergeObservation(Observation observation) {
     ObservationEntity observationEntity = ObservationEntity.fromObservation(observation);
     return observationMutationDao
         .findByObservationId(observation.getId())
-        .flatMapCompletable(mutations -> mergeObservation(observationEntity, mutations));
+        .flatMapCompletable(mutations -> mergeObservation(observationEntity, mutations))
+        .subscribeOn(schedulers.io());
   }
 
   private Completable mergeObservation(
       ObservationEntity observation, List<ObservationMutationEntity> mutations) {
     if (mutations.isEmpty()) {
-      return Completable.complete();
+      return observationDao.insertOrUpdate(observation);
```

Comment:
If we traceback the call, this observation is fetched from local db in `getObservations()` and isn't mutated in the path. Maybe we should revert this back too. Unless I am missing some point, can you please check?

Reply:
Observations are being fetched from the _remote_ db in `ObservationRepository` line 97, not the local one; previous they were not being written at all unless there were related mutations; not sure how this was ever working, unless something changed.

Reply:
Thanks for pointing that out!